app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt

```diff
@@ -118,7 +115,45 @@ class HrAppTimeTrackerPresenter(
     }
 
     override fun onNewTimeReportClicked() {
-        router?.openTimeReportDetail()
+        router?.openCreateTimeReport()
+    }
+
+    override fun onEditTimeReportClicked(reportViewModel: UserReportViewModel) {
+        router?.openEditTimeReport(reportViewModel, reportViewModel.date.toCalendar())
+    }
```

Comment:
@AlexanderKozubets Please, pay your attention on onTaskCreated(), onTaskUpdated() and onTaskDeleted methods.